### PR TITLE
Add docs about converting between shapely and cf

### PR DESCRIPTION
**Pull request by jsignell**
I finally got around to taking a crack at #497. Feel free to add anything.

### CODE REVIEW

**Comment by jsignell** [comment]
Not really sure how coverage can go down on a pure docs PR 😅

File: doc/geometry.md [hunk 1]
```diff
@@ -1,7 +1,98 @@
...
 .. currentmodule:: xarray
 ```
 
 # Geometries
 
-See {py:func}`cf_xarray.shapely_to_cf`,  {py:func}`cf_xarray.cf_to_shapely`
+See
+
+1. {py:func}`cf_xarray.shapely_to_cf`
+1. {py:func}`cf_xarray.cf_to_shapely`
```
Comment (dcherian) [inline doc/geometry.md:19]:
````suggestion
```seealso
1. {py:func}`cf_xarray.shapely_to_cf`
1. {py:func}`cf_xarray.cf_to_shapely`
```
````

File: doc/geometry.md [hunk 2]
```diff
@@ -1,7 +1,98 @@
...
+
+Let's start by creating an xarray object containing some shapely geometries. This example uses
+a `xr.DataArray` but these functions also work with a `xr.Dataset` where one of the data variables
+contains an array of shapes.
+
+```{code-cell}
+import cf_xarray as cfxr
+import xarray as xr
+
+from shapely.geometry import MultiPoint, Point
```
Comment (dcherian) [inline doc/geometry.md:32]: we'll need to add `shapely` to the docs environment.

File: doc/geometry.md [hunk 3]
```diff
@@ -1,7 +1,98 @@
...
+- 'x', 'y' : the node coordinates
+- 'crd_x', 'crd_y' : the feature coordinates (might have different names if `grid_mapping` is available).
+- 'node_count' : The number of nodes per feature. Always present for Lines and Polygons. For
+  Points: only present if there are multipart geometries.
+- part_node_count : The number of nodes per individual geometry. Only for Lines with multipart
+  geometries and for Polygons with multipart geometries or holes.
+- interior_ring : Integer boolean indicating whether ring is interior or exterior. Only for
+  Polygons with holes.
+- 'geometry_container' : Empty variable with attributes describing the geometry type.
+
```
Comment (dcherian) [inline doc/geometry.md:71]:
```suggestion
- `'x'`, `'y'` : the node coordinates
- `'crd_x'`, `'crd_y'` : the feature coordinates (might have different names if `grid_mapping` is available).
- `'node_count'` : The number of nodes per feature. Always present for Lines and Polygons. For
  Points: only present if there are multipart geometries.
- `'part_node_count'` : The number of nodes per individual geometry. Only for Lines with multipart
  geometries and for Polygons with multipart geometries or holes.
- `'interior_ring'` : Integer boolean indicating whether ring is interior or exterior. Only for
  Polygons with holes.
- `'geometry_container`' : Empty variable with attributes describing the geometry type.

```

File: doc/geometry.md [hunk 4]
```diff
@@ -1,7 +1,99 @@
...
+---
+
 ```{eval-rst}
 .. currentmodule:: xarray
 ```
 
 # Geometries
 
-See {py:func}`cf_xarray.shapely_to_cf`,  {py:func}`cf_xarray.cf_to_shapely`
+```seealso
```
Comment (dcherian) [inline doc/geometry.md:16]:
```suggestion
```{seealso}
```

File: doc/geometry.md [hunk 5]
```diff
@@ -1,7 +1,99 @@
...
 
-See {py:func}`cf_xarray.shapely_to_cf`,  {py:func}`cf_xarray.cf_to_shapely`
+```{seealso}
+1. [The CF conventions on Geometries](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#geometries)
+1. {py:func}`cf_xarray.shapely_to_cf`
+1. {py:func}`cf_xarray.cf_to_shapely`
+```
+
+In order to support vectors as well as arrays `cf_xarray` can convert between shapely objects
+and CF-compliant representations of those geometries.
```
Comment (dcherian) [inline doc/geometry.md:23]:
```suggestion
`cf_xarray` can convert between vector geometries represented as shapely objects
and CF-compliant array representations of those geometries.
```

**Comment by dcherian** [comment]
Thanks. Great improvement over an empty page :D

**Comment by jsignell** [comment]
Now there is a page about points :)

**Pull request merged**